stl/inc/atomic

```diff
@@ -418,21 +418,42 @@ void _Atomic_wait_direct(
 }
 #endif // _HAS_CXX20
 
-#if 1 // TRANSITION, ABI
+#if 1 // TRANSITION, ABI, GH-1151
 inline void _Atomic_lock_acquire(long& _Spinlock) noexcept {
-    while (_InterlockedExchange(&_Spinlock, 1)) {
-        _YIELD_PROCESSOR();
+#if defined(_M_IX86) || defined(_M_X64)
+    // Algorithm from Intel(R) 64 and IA-32 Architectures Optimization Reference Manual, September 2019
+    // Example 2-4. Contended Locks with Increasing Back-off Example - Improved Version, page 2-21

```

Comment:
```suggestion
    // Algorithm from Intel(R) 64 and IA-32 Architectures Optimization Reference Manual, May 2020
    // Example 2-4. Contended Locks with Increasing Back-off Example - Improved Version, page 2-22

```

Alternately we should drop this reference completely; it's not like the manual saying 'you should implement exponential backoff' is particularly novel.